src/saltext/vmware/states/esxi.py

```diff
@@ -1,3 +1,216 @@
-"""
-Nothing is here yet, this is a placeholder to be able to build docs.
-"""
+# SPDX-License-Identifier: Apache-2.0
+import logging
+
+from saltext.vmware.utils.connect import get_service_instance
+
+
+log = logging.getLogger(__name__)
+
+try:
+    import pyVmomi
+
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+
+__virtualname__ = "vmware_esxi"
+__proxyenabled__ = ["vmware_esxi"]
+
+
+def __virtual__():
+    if not HAS_PYVMOMI:
+        return False, "Unable to import pyVmomi module."
+    return __virtualname__
+
+
+def user_present(
+    name,
+    password,
+    description=None,
+    datacenter_name=None,
+    cluster_name=None,
+    host_name=None,
+    service_instance=None,
+):
+    """
+    Add local user on matching ESXi hosts.
+
+    name
+        User to create on matching ESXi hosts. (required).
+
+    password
+        Password for the user. (required).
+
+    description
+        Description of the user. (optional).
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: yaml
+
+    Create User:
+      vmware_esxi.user_present:
+        - name: local_user
+        - password: secret
+
+    """
+    log.debug("Running vmware_esxi.user_present")
+    ret = {"name": name, "result": None, "comment": "", "changes": {}}
+    create = update = 0
+    diff = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    user = __salt__["vmware_esxi.get_user"](
+        user_name=name,
+        datacenter_name=datacenter_name,
+        cluster_name=cluster_name,
+        host_name=host_name,
+        service_instance=service_instance,
+    )
+    hosts = __salt__["vmware_esxi.list_hosts"](
+        datacenter_name=datacenter_name,
+        cluster_name=cluster_name,
+        host_name=host_name,
+        service_instance=service_instance,
+    )
+    for host in hosts:
+        if host in user:
```

Comment:
This line is hard to read

 `user = dict{host: {user}}`, so it works, but even knowing what it was it was unexpected. 

Maybe a name other than `user` would be good for this variable, something like `users_by_host`?

Reply:
sure will change